tests/test_singularity.py

```diff
@@ -252,3 +258,150 @@ def test_singularity_dockerfile_with_name_with_cache(
     assert not (cachedir / "bea92b9b6910cbbd2ae602f5bb0f0f27_latest.sif").exists()
     assert not (workdir / "customDebian_latest.sif").exists()
     assert (cachedir / "customDebian_latest.sif").exists()
+
+
+@needs_singularity
+def test_singularity_local_sandbox_image(tmp_path: Path) -> None:
+    workdir = tmp_path / "working_dir"
+    workdir.mkdir()
+    # build a sandbox image
+    container_path = workdir / "container_repo"
+    container_path.mkdir()
+    cmd = [
+        "singularity",
+        "build",
+        "--sandbox",
+        str(container_path / "alpine"),
+        "docker://alpine:latest",
+    ]
+    build = subprocess.run(cmd, capture_output=True, text=True)
+    if build.returncode == 0:
+        # test that we can work in sub directories
+        with working_directory(workdir):
+            result_code, _, _ = get_main_output(
+                [
+                    "--singularity",
+                    "--disable-pull",
+                    get_data("tests/sing_local_sandbox_test.cwl"),
+                    "--message",
+                    "hello",
+                ]
+            )
+            assert result_code == 0
+            result_code, _, _ = get_main_output(
+                [
+                    "--singularity",
+                    "--disable-pull",
+                    get_data("tests/sing_local_sandbox_img_id_test.cwl"),
+                    "--message",
+                    "hello",
+                ]
+            )
+            assert result_code == 0
+        # test with --singularity-sandbox-path option:
+        result_code, out, err = get_main_output(
+            [
+                "--singularity",
+                "--disable-pull",
+                "--singularity-sandbox-path",
+                f"{workdir}",
+                get_data("tests/sing_local_sandbox_test.cwl"),
+                "--message",
+                "hello",
+            ]
+        )
+        assert result_code == 0
+
+        # test with CWL_SINGULARITY_IMAGES env variable set:
+        os.environ["CWL_SINGULARITY_IMAGES"] = str(workdir)
+        result_code, _, _ = get_main_output(
+            [
+                "--singularity",
+                "--disable-pull",
+                get_data("tests/sing_local_sandbox_test.cwl"),
+                "--message",
+                "hello",
+            ]
+        )
+        assert result_code == 0
```

Comment:
Please use a monkeypatch to adjust the environ, needed for parallel execution of tests
https://docs.pytest.org/en/stable/how-to/monkeypatch.html#monkeypatching-environment-variables